kit/src/lib/FrameworkContentBlock.svelte

```diff
@@ -0,0 +1,76 @@
+<script lang="ts">
+	import type { SvelteComponent } from "svelte";
+	import type { Framework } from "./types";
+
+	import { onMount } from "svelte";
+	import { getFrameworkStore } from "./stores";
+	import IconPytorch from "./IconPytorch.svelte";
+	import IconTensorflow from "./IconTensorflow.svelte";
+	import IconJax from "./IconJax.svelte";
+	import IconEyeShow from "./IconEyeShow.svelte";
+	import IconEyeHide from "./IconEyeHide.svelte";
+
+	export let framework: Framework;
+
+	const FRAMEWORK_CONFIG: Record<Framework, { Icon: typeof SvelteComponent; label: string }> = {
+		pytorch: {
+			Icon: IconPytorch,
+			label: "Pytorch"
+		},
+		tensorflow: {
+			Icon: IconTensorflow,
+			label: "TensorFlow"
+		},
+		jax: {
+			Icon: IconJax,
+			label: "JAX"
+		}
+	};
+	const { Icon, label } = FRAMEWORK_CONFIG[framework];
+	const localStorageKey = `hf_doc_framework_${framework}_is_hidden`;
+	const frameworkStore = getFrameworkStore(framework);
+
+	function toggleHidden() {
+		$frameworkStore = !$frameworkStore;
+		localStorage.setItem(localStorageKey, $frameworkStore ? "true" : "false");
+	}
+
+	onMount(() => {
+		if (localStorage.getItem(localStorageKey) === "true") {
+			$frameworkStore = true;
+		}
+	});
```

Comment:
maybe it could be stored in DB (ie. only be a feature for logged in user) the same way we store dark mode (just an idea)

Reply:
on top of my head, not sure how we can pass this info to https://github.com/huggingface/moon-landing/blob/master/server%2Fviews%2Fpages%2FDocPage%2FDocBuilderPage.svelte#L54

cc: @coyotte508

Reply:
one possibility is (maybe, not sure) to use svelte `context` for that, but rather convoluted - let's keep that for a second time :)

Reply:
> one possibility is (maybe, not sure) to use svelte `context` for that, but rather convoluted

Would have never thought of that, but if the JS is compatible between the generated doc & the site maybe =o

But old generated version could break if svelte is upgraded.

I think maybe something with `window.postMessage` to trigger the doc -> moon-landing update and a `data-hide-x-framework` on `<body>` (or `<meta>` in `svelte:head`...) to get the initial value?

What about logged out users though? `localStorage` works better for them

> let's keep that for a second time :)

Definitely!